build/packages-template/bbb-web/bbb-process-image.sh

```diff
@@ -0,0 +1,132 @@
+
+echo "Starting bbb-process-image"
+
+PAGE_NUMBER=${PAGE_NUMBER-1}
+
+# Define directory paths and file locations
+BASE_DIR="${PRESENTATIONS_DIR}/${MEETING_ID}/${MEETING_ID}/${PRES_ID}"
+IMAGE_FILE="${BASE_DIR}/${PRES_ID}_${PAGE_NUMBER}.${IMG_FORMAT}"
+IMAGE_FILE_WITHOUT_PAGE="${BASE_DIR}/${PRES_ID}.${IMG_FORMAT}"
+TEXT_DIR="${BASE_DIR}/textfiles"
+THUMBNAIL_DIR="${BASE_DIR}/thumbnails"
+PNG_DIR="${BASE_DIR}/pngs"
+SVG_DIR="${BASE_DIR}/svgs"
+
+THUMBNAIL_OUTPUT="${THUMBNAIL_DIR}/thumb-${PAGE_NUMBER}.png"
+TEXT_OUTPUT="${TEXT_DIR}/slide-${PAGE_NUMBER}.txt"
+
+# when it's page one maybe the filename doesn't contain the page
+if [ "$PAGE_NUMBER" = "1" ] && [ ! -e "$IMAGE_FILE" ] && [ -e "$IMAGE_FILE_WITHOUT_PAGE" ]; then
+  IMAGE_FILE=$IMAGE_FILE_WITHOUT_PAGE
+elif [ ! -e "$IMAGE_FILE" ]; then
+    echo "Image not found: $IMAGE_FILE"
+    exit 1
+fi
+
+# Write default text for the slide
+if [ ! -e $TEXT_OUTPUT ]; then
+  echo "No text could be retrieved for the slide" >"${TEXT_OUTPUT}"
+fi
+
+
+# Check image width and height
+IFS=' ' read IMAGE_WIDTH IMAGE_HEIGHT <<<"$(identify -format "%w %h" "${IMAGE_FILE}")" || {
+  echo "Error extracting image dimensions"
+  exit 1
+}
+
+if [ "$IMAGE_WIDTH" -gt "$MAX_IMAGE_WIDTH" ] || [ "$IMAGE_HEIGHT" -gt "$MAX_IMAGE_HEIGHT" ]; then
+  echo "The image exceeds max dimension allowed, it will be resized"
+  convert -resize "${MAX_IMAGE_WIDTH}x${MAX_IMAGE_HEIGHT}" "${IMAGE_FILE}" "${IMAGE_FILE}"
+fi
+
+# Create a thumbnail using ImageMagick
+/usr/bin/convert -thumbnail "${THUMBNAIL_SCALE}x${THUMBNAIL_SCALE}" "${IMAGE_FILE}" "${THUMBNAIL_OUTPUT}"
+
+# Convert image to PDF for SVG conversion (and perhaps for png)
+PDF_FOR_SVG_OR_PNG="${BASE_DIR}/slide-${PAGE_NUMBER}.png.pdf"
+if ! /usr/bin/convert "$IMAGE_FILE" -auto-orient "$PDF_FOR_SVG_OR_PNG"; then
+  echo "PDF conversion for SVG generation failed."
+  exit 1
+fi
+
+# If PNG generation is enabled, convert the image to PDF then to PNG
+if [ "${GENERATE_PNGS}" = "true" ]; then
+  echo "generatePngs flag enabled"
+  /usr/bin/pdftocairo -png -scale-to "${PNG_SCALE_TO}" -f 1 -l 1 "$PDF_FOR_SVG_OR_PNG" "${PNG_DIR}/slide"
+  # Pdf cairo saves as slide-01.png, and BBB expects slide-1.png, renaming
+  mv "${PNG_DIR}/slide-0${PAGE_NUMBER}.png" "${PNG_DIR}/slide-${PAGE_NUMBER}.png" 2>/dev/null || true
+fi
+
+# Define output SVG file
+SVG_OUTPUT="${SVG_DIR}/slide${PAGE_NUMBER}.svg"
+
+# Create a temporary file for PNG conversion and ensure cleanup on exit
+TEMP_PNG=$(mktemp)
+trap 'rm -f "${TEMP_PNG}"*' EXIT
+
+echo "Converting PDF to PNG for rasterized SVG creation."
+pdftocairo -r ${SVG_RESOLUTION_PPI} -png -singlefile -scale-to-x ${RASTERIZE_PNG_WIDTH} -scale-to-y -1 -q -f 1 -l 1 ${PDF_FOR_SVG_OR_PNG} ${TEMP_PNG}
+
+PNG_FILE="${TEMP_PNG}.png"
+png_size=$(stat -c%s "${PNG_FILE}" 2>/dev/null || echo 0)
+
+if [ ! "$png_size" -gt 0 ]; then
+    echo "PNG conversion did not produce a valid file."
+    exit 1
+fi
+
+base64_png=$(base64 -w 0 "${PNG_FILE}")
```

Comment:
I recommend checking the size of the png file *before* base64 encoding. Note that the size expansion caused by base64 is a fixed ratio of 133% (since line-wrapping is disabled).